File: src/eval_workbench/frontend/src/utils/snapshotLabel.ts
export interface SnapshotLike {
  id: string;
  commit_hash?: string;
  agent_target?: {
    agent_path?: string;
    name?: string;
  };
}

export function shortCommitHash(commitHash: string, length = 7): string {
  if (!commitHash) return '';
  return commitHash.length > length ? commitHash.slice(0, length) : commitHash;
}

/** Human-readable snapshot label: short_hash:module.path:agent_var */
export function formatSnapshotLabel(snapshot: SnapshotLike): string {
  const agentPath = snapshot.agent_target?.agent_path;
  const commit = snapshot.commit_hash || snapshot.id.split(':')[0] || '';
  const shortHash = shortCommitHash(commit);

  if (agentPath) {
    return `${shortHash}:${agentPath}`;
  }

  // Fallback: shorten commit prefix of canonical id (commit:module:var)
  const colon = snapshot.id.indexOf(':');
  if (colon > 0) {
    const idCommit = snapshot.id.slice(0, colon);
    const rest = snapshot.id.slice(colon + 1);
    return `${shortCommitHash(idCommit)}:${rest}`;
  }

  return snapshot.id;
}

File: src/eval_workbench/services/snapshot_label.py
"""Human-readable snapshot labels for UI and APIs."""

from __future__ import annotations

from typing import Any


def short_commit_hash(commit_hash: str, length: int = 7):
    if not commit_hash:
        return ""
    return commit_hash[:length] if len(commit_hash) > length else commit_hash


def format_snapshot_label(snapshot: dict[str, Any]):
    """Format as short_hash:module.path:agent_var."""
    agent_target = snapshot.get("agent_target") or {}
    agent_path = agent_target.get("agent_path")
    commit = snapshot.get("commit_hash") or snapshot.get("id", "").split(":", 1)[0]
    short_hash = short_commit_hash(commit)

    if agent_path:
        return f"{short_hash}:{agent_path}"

    snapshot_id = snapshot.get("id", "")
    if ":" in snapshot_id:
        id_commit, rest = snapshot_id.split(":", 1)
        return f"{short_commit_hash(id_commit)}:{rest}"

    return snapshot_id
